### PR TITLE
feat: change liquidation mechanism to transfer all

### PR DESCRIPTION
**Pull request by kjr217**
# Task: Full liquidation mechanism

## High Level Description

+ Change liquidation mechanism so it fully transfers vault collateral to liquidator without an auction once the vault goes underwater

### CODE REVIEW

File: contracts/core/MarginCalculator.sol [hunk 1]
```diff
@@ -990,54 +957,11 @@ contract MarginCalculator is Ownable {
         FPI.FixedPointInt memory _vaultDebt,
         FPI.FixedPointInt memory _cashValue,
```
Comment (antoncoding) [inline contracts/core/MarginCalculator.sol:958]: we don't need cashValue and spotPrice either.

File: contracts/tests/CalculatorTester.sol [hunk 2]
```diff
@@ -55,8 +55,7 @@ contract CalculatorTester is MarginCalculator {
         FixedPointInt256.FixedPointInt memory vaultDebt = FixedPointInt256.fromScaledUint(_vaultDebt, BASE);
         FixedPointInt256.FixedPointInt memory cv = FixedPointInt256.fromScaledUint(_cv, BASE);
         FixedPointInt256.FixedPointInt memory spotPrice = FixedPointInt256.fromScaledUint(_spotPrice, BASE);
-        OptionType opType = getOptionType(_isPut, collateral, underlying);
-        return
-            _getDebtPrice(vaultCollateral, vaultDebt, cv, spotPrice, _auctionStartingTime, _collateralDecimals, opType);
+        OptionType opType = getOptionType(_isPut, underlying, collateral);
+        return _getDebtPrice(vaultCollateral, vaultDebt, cv, spotPrice, _collateralDecimals, opType);
```
Comment (antoncoding) [inline contracts/tests/CalculatorTester.sol:59]: don't need cv and spotPrice

**Review by antoncoding — APPROVED** [review]
Can remove more logic, other than that, looks good

File: contracts/core/MarginCalculator.sol (hunk 1 above)
Comment (kjr217) [reply contracts/core/MarginCalculator.sol:958]: Im pretty sure you need cashValue

**Review by antoncoding — APPROVED** [review]

**Pull request merged**